# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/02 09:49:29-08:00 rddunlap@osdl.org 
#   [PATCH] parport_pc: don't mix module parameter styles
#   
#   Somehow parport_pc.c ended up with mixed old-style and new-style module
#   parameters, but mixing them is not allowed.
#   
#   Use module_param() instead of MODULE_PARM() -- cannot be mixed.
#   
#   Signed-off-by: Randy Dunlap <rddunlap@osdl.org>
#   Signed-off-by: Andrew Morton <akpm@osdl.org>
#   Signed-off-by: Linus Torvalds <torvalds@osdl.org>
# 
# drivers/parport/parport_pc.c
#   2005/01/02 01:37:36-08:00 rddunlap@osdl.org +2 -2
#   parport_pc: don't mix module parameter styles
# 
diff -Nru a/drivers/parport/parport_pc.c b/drivers/parport/parport_pc.c
--- a/drivers/parport/parport_pc.c	2005-01-04 18:24:07 -08:00
+++ b/drivers/parport/parport_pc.c	2005-01-04 18:24:07 -08:00
@@ -3176,7 +3176,6 @@
 #ifdef MODULE
 static const char *irq[PARPORT_PC_MAX_PORTS];
 static const char *dma[PARPORT_PC_MAX_PORTS];
-static char *init_mode;
 
 MODULE_PARM_DESC(io, "Base I/O address (SPP regs)");
 module_param_array(io, int, NULL, 0);
@@ -3192,8 +3191,9 @@
 module_param(verbose_probing, int, 0644);
 #endif
 #ifdef CONFIG_PCI
+static char *init_mode;
 MODULE_PARM_DESC(init_mode, "Initialise mode for VIA VT8231 port (spp, ps2, epp, ecp or ecpepp)");
-MODULE_PARM(init_mode, "s");
+module_param(init_mode, charp, 0);
 #endif
 
 static int __init parse_parport_params(void)
